pkg/controller/operators/openshift/options.go

```diff
@@ -57,7 +57,7 @@ func (c *ReconcilerConfig) complete() error {
 		return fmt.Errorf("no ClusterOperator name specified")
 	}
 	if c.Log == nil {
-		c.Log = ctrl.Log.WithName(c.Name)
+		c.Log = ctrl.Log.WithName(c.Name).GetSink()
```

Comment:
Just for my own edification: this change is needed since the logr.Logger type has changed, and we can no longer nil check it's value, or was there some other breaking change in the logr major release bump?

Reply:
"The breaking change in logr seems to be that logr.Logger is an interface in v0.4.0, renamed to logr.LogSink, and a new type logr.Logger is defined as a struct."
https://github.com/kubernetes-sigs/controller-runtime/issues/1607
This lead to " invalid operation: logging.logr == nil (mismatched types logr.Logger and nil"

Reply:
I think we can minimize the changes to this file but only updating the conditional check on line 59 without mucking with any updated type declarations. See https://github.com/operator-framework/operator-lifecycle-manager/pull/2574/files#r789780717 for what I had in mind.